### PR TITLE
Add description for arg_parse

### PR DESCRIPTION
**Pull request by anu0012**
This PR fixes https://github.com/coala/coala/issues/2811

### CODE REVIEW

**Comment by Udayan12167** [comment]
git commit shortlog is incorrect. It should be of the format `ConfigurationGathering: Add description for arg_parse`. It should always be in imperative tense and shorter than 50 chars.

**Comment by sils1297** [comment]
issue reference missing, see coala.io/commit

File: coalib/settings/ConfigurationGathering.py [hunk 1]
```diff
@@ -129,7 +129,8 @@ def load_configuration(arg_list, log_printer, arg_parser=None):
 
     :param arg_list:    The list of command line arguments.
     :param log_printer: The LogPrinter object for logging.
-    :param arg_parser
+    :param arg_parser:  Instance of ArgParser that is used to
+                        parse none-setting arguments.
```
Comment (adtac) [inline coalib/settings/ConfigurationGathering.py]: this should be in the previous commit itself, right?

Comment (anu0012) [inline coalib/settings/ConfigurationGathering.py]: Yes, I know. Sorry about that. Can you tell me which branch to delete and how to delete it ?

**Pull request closed without merging**